packages/sources/1forge/src/endpoint/quotes.ts

```diff
@@ -0,0 +1,79 @@
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+//import { warmupStopped } from '@chainlink/ea-bootstrap/src/lib/cache-warmer/actions'
+import { ExecuteWithConfig, Config, InputParameters, AdapterRequest, AxiosResponse } from '@chainlink/types'
+import { NAME as AdapterName } from '../config'
+
+export const supportedEndpoints = ['quotes']
+export const batchablePropertyPath = ['base', 'quote']
+
+export const inputParameters: InputParameters = {
+  base: ['base', 'from'],
+  quote: ['quote', 'to'],
+  quantity: false,
+}
+
+const handleBatchedRequest = (
+  jobRunID: string,
+  request: AdapterRequest,
+  response: AxiosResponse<any>,
+  resultPath: string,
+) => {
+  const payload: [AdapterRequest, number][] = []
+  for (const pair of response.data) {
+    const base = pair['s'].split('/')[0]
+    const quote = pair['s'].split('/')[1]
+    payload.push([
+      {
+        ...request,
+        data: { ...request.data, base: base.toUpperCase(), quote: quote.toUpperCase() },
+      },
+      Requester.validateResultNumber(pair, [resultPath]),
+    ])
+
+  }
+  return Requester.success(jobRunID, Requester.withResult(response, undefined, payload), true, ['base', 'quote'])
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+  const jobRunID = validator.validated.id
+  const url = `/quotes`
+  const from = validator.overrideSymbol(AdapterName)
+  const to = validator.validated.data.quote
+  const pairArray = []
+
+  for (const fromCurrency of formatArray(from)) {
+    for (const toCurrency of formatArray(to)) {
+      pairArray.push(`${fromCurrency.toUpperCase()}/${toCurrency.toUpperCase()}`)
```

Comment:
What if to is just a string?

Individual feed requests come in as: to: 'USD'.

We need to handle both arrays and strings.

Reply:
The `formatArray` function makes it so both strings and arrays are handled. If its an array, it returns the array, if its a string it makes it into an array w/ one string.

Reply:
Nice. 
One suggestion for something like this would be to throw it in the core framework's `utils` to re-use it across adapters.

Reply:
Did this in most recent commit